providers/PointsProvider.tsx

```diff
@@ -0,0 +1,19 @@
+'use client'
+
+import React, { createContext, useContext } from 'react'
+
+const PointsContext = createContext(undefined)
+
+export function PointsProvider({ events, leaderboardRank, children }) {
+  return (
+    <PointsContext.Provider value={{ events, leaderboardRank }}>{children}</PointsContext.Provider>
+  )
+}
+
+export function usePoints() {
+  const context = useContext(PointsContext)
+  if (!context) {
+    throw new Error('usePointsProvider must be used within a PointsProvider')
```

Comment:
function name should match error being thrown
```suggestion
export function usePointsProvider() {
  const context = useContext(PointsContext)
  if (!context) {
    throw new Error('usePointsProvider must be used within a PointsProvider')
```